PROTO_tests/tests/indexing_test.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+
+NUM_TYPES = [ak.int64, ak.uint64, ak.float64, ak.bool, ak.bigint]
+
+
+def key_arrays(size):
+    ikeys = ak.arange(size)
+    ukeys = ak.arange(size, dtype=ak.uint64)
+    return ikeys, ukeys
+
+
+def value_array(dtype, size):
+    if dtype in [ak.int64, ak.float64]:
+        return ak.randint(-size, size, size, dtype=dtype)
+    elif dtype is ak.uint64:
+        return ak.randint(0, size, size, dtype=dtype)
+    elif dtype is ak.bool:
+        return (ak.randint(0, size, size) % 2) == 0
+    elif dtype is ak.bigint:
+        return ak.randint(0, size, size, dtype=ak.uint64) + 2**200
+    elif dtype is ak.str_:
+        return ak.random_strings_uniform(1, 16, size=size)
+    return None
+
+
+def value_scalar(dtype, size):
+    if dtype in [ak.int64, ak.float64]:
+        return ak.randint(-size, 0, 1, dtype=dtype)
+    elif dtype is ak.uint64:
+        return ak.randint(2**63, 2**64, 1, dtype=dtype)
+    elif dtype is ak.bool:
+        return (ak.randint(0, 2, 1) % 2) == 0
+    elif dtype is ak.bigint:
+        return ak.randint(0, size, 1, dtype=ak.uint64) + 2**200
+    elif dtype is ak.str_:
+        return ak.random_strings_uniform(1, 16, size=1)
+    return None
+
+
+class TestIndexing:
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", NUM_TYPES + [ak.str_])
+    def test_pdarray_uint_indexing(self, prob_size, dtype):
+        ikeys, ukeys = key_arrays(prob_size)
+        pda = value_array(dtype, prob_size)
+        assert pda[np.uint(2)] == pda[2]
+        assert pda[ukeys].to_list() == pda[ikeys].to_list()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_bool_indexing(self, prob_size):
+        u = value_array(ak.uint64, prob_size)
+        b = value_array(ak.bool, prob_size)
+        assert u[b].to_list() == ak.cast(u, ak.int64)[b].to_list()
+        assert u[b].to_list() == ak.cast(u, ak.bigint)[b].to_list()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", NUM_TYPES)
+    def test_set_uint(self, prob_size, dtype):
```

Comment:
this test fails, prob because we set `pda[ak.arange(prob_size)] = value_array(dtype, prob_size)`, but test that `pda[:prob_size].to_list() == ak.arange(prob_size, dtype=dtype).to_list()`

Reply:
Updated